[DO NOT MERGE]  make post-migration check happy transforming failing assertions in to log errors

sigurpol opened the pull request:
Transform all assertions that cause the post-migration check to fail during the AHM dry run into log errors.

Command used to test:

```bash
RUST_LOG=runtime::ah-migrator=debug,runtime::rc-migrator=debug,remote-ext=info,runtime=warn,runtime=info SNAP_RC_PRE="/Users/paolo/github/ahm-dryrun/runtimes/snaps/kusama-rc-pre.snap" SNAP_AH_PRE="/Users/paolo/github/ahm-dryrun/runtimes/snaps/kusama-ah-pre.snap" SNAP_RC_POST="/Users/paolo/github/ahm-dryrun/runtimes/snaps/kusama-rc-post.snap" SNAP_AH_POST="/Users/paolo/github/ahm-dryrun/runtimes/snaps/kusama-ah-post.snap" cargo test -p polkadot-integration-tests-ahm --release --features kusama-ahm --features try-runtime post_migration_checks_only -- --include-ignored --nocapture --test-threads 1
```

[test.log](https://github.com/user-attachments/files/22516505/test.log)

kianenigma commented on integration-tests/ahm/src/checks.rs:
```diff
@@ -43,10 +43,10 @@ impl RcMigrationCheck for SanityChecks {
-			pallet_rc_migrator::RcMigrationStage::<RcRuntime>::get(),
-			pallet_rc_migrator::MigrationStage::MigrationDone
-		);
+		// assert_eq!(
```
yeah this one was stuck at `MigrationSignal` or similar, probably because our snapshot, for whatever reason, is still not taken at the right time :(

pepoviola replied:
I tested locally with the _snaps_ from [this run](https://github.com/paritytech/ahm-dryrun/actions/runs/17954044356/job/51060475655) and works as expected.

ggwpez commented on pallets/ah-migrator/src/staking/checks.rs:
```diff
@@ -214,6 +214,14 @@ fn assert_equal_items<
 	ah.sort_by_encoded();
 
 	for (i, (r, a)) in rc.iter().zip(ah.iter()).enumerate() {
-		assert_eq!(r, a, "Entry #{i} mismatch: {r:?} != {a:?}");
```
Wondering why this could fail?

ggwpez replied:
Maybe authorship rewards

sigurpol commented:
Retested with current latest main (c2a76dac06d838299cb3af387d1ba0c7b2f66b3d) and snapshots from local build.
Proxy issue doesn't happen any more, same for one in society.rs  and skip check vs MigrationDone.

To make it pass now I commented out the following assertion in the following files:
```bash
	modified:   pallets/ah-migrator/src/account.rs
	modified:   pallets/ah-migrator/src/scheduler.rs
	modified:   pallets/ah-migrator/src/staking/bags_list.rs 
	modified:   pallets/ah-migrator/src/staking/checks.rs
	modified:   pallets/rc-migrator/src/scheduler.rs
```

```bash
diff --git a/pallets/ah-migrator/src/account.rs b/pallets/ah-migrator/src/account.rs
index f59d2494d..228375bd9 100644
--- a/pallets/ah-migrator/src/account.rs
+++ b/pallets/ah-migrator/src/account.rs
@@ -379,11 +379,11 @@ pub mod tests {
 				// Therefore, we just check that the difference between the balance migrated from
 				// the RC to AH and the balance delta on AH before and after migration is less than
 				// AH existential deposit.
-				assert!(
-					rc_migrated_balance.saturating_sub(ah_migrated_balance) < ah_ed,
-					"Total balance mismatch for account {:?} between RC pre-migration and AH post-migration",
-					who.to_ss58check()
-				);
+				// assert!(
+				// rc_migrated_balance.saturating_sub(ah_migrated_balance) < ah_ed,
+				// "Total balance mismatch for account {:?} between RC pre-migration and AH
+				// post-migration", who.to_ss58check()
+				// );
 
 				// There are several `unreserve` operations on AH after migration (e.g., unreserve
 				// deposits for multisigs because they are not migrated to AH, adjust deposits for
@@ -391,7 +391,7 @@ pub mod tests {
 				// AH after migration is less than the migrated reserved balance from RC.
 				assert!(
 					ah_reserved_post.saturating_sub(ah_reserved_before) <= summary.migrated_reserved,
-					"Change in reserved balance on AH after migration for account {:?} is greater than the migrated reserved balance from RC", 
+					"Change in reserved balance on AH after migration for account {:?} is greater than the migrated reserved balance from RC",
 					who.to_ss58check()
 				);
 
diff --git a/pallets/ah-migrator/src/scheduler.rs b/pallets/ah-migrator/src/scheduler.rs
index 528e99975..93b6fb8da 100644
--- a/pallets/ah-migrator/src/scheduler.rs
+++ b/pallets/ah-migrator/src/scheduler.rs
@@ -195,11 +195,11 @@ impl<T: Config> crate::types::AhMigrationCheck for SchedulerMigrator<T> {
 
 		// Assert storage 'Scheduler::IncompleteSince::ah_post::correct'
 		if rc_payload.incomplete_since.is_some() {
-			assert_eq!(
-				pallet_scheduler::IncompleteSince::<T>::get(),
-				rc_payload.incomplete_since,
-				"IncompleteSince on Asset Hub should match the RC value"
-			);
+			// assert_eq!(
+			// pallet_scheduler::IncompleteSince::<T>::get(),
+			// rc_payload.incomplete_since,
+			// "IncompleteSince on Asset Hub should match the RC value"
+			// );
 		} else {
 			assert_eq!(
 				pallet_scheduler::IncompleteSince::<T>::get(),
diff --git a/pallets/ah-migrator/src/staking/bags_list.rs b/pallets/ah-migrator/src/staking/bags_list.rs
index 042761d8a..684753536 100644
--- a/pallets/ah-migrator/src/staking/bags_list.rs
+++ b/pallets/ah-migrator/src/staking/bags_list.rs
@@ -203,10 +203,10 @@ impl<T: Config> crate::types::AhMigrationCheck for BagsListMigrator<T> {
 		// Assert storage "VoterList::ListNodes::ah_post::consistent"
 		// Assert storage "VoterList::ListBags::ah_post::correct"
 		// Assert storage "VoterList::ListBags::ah_post::consistent"
-		assert_eq!(
-			rc_pre_translated, ah_messages,
-			"Bags list data mismatch: Asset Hub data differs from original Relay Chain data"
-		);
+		// assert_eq!(
+		// rc_pre_translated, ah_messages,
+		// "Bags list data mismatch: Asset Hub data differs from original Relay Chain data"
+		// );
 
 		// Run bags-list pallet integrity check
 		#[cfg(feature = "try-runtime")]
diff --git a/pallets/ah-migrator/src/staking/checks.rs b/pallets/ah-migrator/src/staking/checks.rs
index 762df4220..fa8bfe564 100644
--- a/pallets/ah-migrator/src/staking/checks.rs
+++ b/pallets/ah-migrator/src/staking/checks.rs
@@ -18,202 +18,269 @@
 
 use codec::Encode;
 use pallet_rc_migrator::{
-	staking::{
-		message::{PortableNominations, PortableUnappliedSlash},
-		RcData,
-	},
-	types::{SortByEncoded, TranslateAccounts},
+    staking::{
+        message::{PortableNominations, PortableUnappliedSlash},
+        RcData,
+    },
+    types::{SortByEncoded, TranslateAccounts},
 };
 use sp_runtime::{AccountId32, Perbill, WeakBoundedVec};
 use std::fmt::Debug;
 
 impl<T: crate::Config> crate::types::AhMigrationCheck
-	for pallet_rc_migrator::staking::StakingMigratedCorrectly<T>
+    for pallet_rc_migrator::staking::StakingMigratedCorrectly<T>
 {
-	type RcPrePayload = RcData;
-	type AhPrePayload = ();
-
-	fn pre_check(_rc: Self::RcPrePayload) -> Self::AhPrePayload {}
-
-	fn post_check(rc: Self::RcPrePayload, _ah_pre_payload: Self::AhPrePayload) {
-		let t = crate::Pallet::<T>::translate_account_rc_to_ah;
-
-		// Storage Values
-		assert_eq!(rc.validator_count, pallet_staking_async::ValidatorCount::<T>::get());
-		// Min validator count is not migrated and instead configured via `MinimumValidatorSetSize`
-		assert_eq!(rc.min_nominator_bond, pallet_staking_async::MinNominatorBond::<T>::get());
-		assert_eq!(rc.min_validator_bond, pallet_staking_async::MinValidatorBond::<T>::get());
-		assert_eq!(rc.min_active_stake, pallet_staking_async::MinimumActiveStake::<T>::get());
-		assert_eq!(rc.min_commission, pallet_staking_async::MinCommission::<T>::get());
-		assert_eq!(rc.max_validators_count, pallet_staking_async::MaxValidatorsCount::<T>::get());
-		assert_eq!(rc.max_nominators_count, pallet_staking_async::MaxNominatorsCount::<T>::get());
-		assert_eq!(
-			pallet_staking_async::CurrentEra::<T>::get().expect("Must be set"),
-			pallet_staking_async::ActiveEra::<T>::get().expect("Must be set").index
-		);
-		assert_eq!(
-			rc.active_era.map(translate_active_era),
-			pallet_staking_async::ActiveEra::<T>::get()
-		);
-		assert_eq!(translate_forcing(rc.force_era), pallet_staking_async::ForceEra::<T>::get());
-		assert_eq!(rc.max_staked_rewards, pallet_staking_async::MaxStakedRewards::<T>::get());
-		assert_eq!(rc.slash_reward_fraction, pallet_staking_async::SlashRewardFraction::<T>::get());
-		assert_eq!(rc.canceled_slash_payout, pallet_staking_async::CanceledSlashPayout::<T>::get());
-		// Current planned session is not migrated
-		assert_eq!(rc.chill_threshold, pallet_staking_async::ChillThreshold::<T>::get());
-
-		// Storage Maps
-		assert_equal_items(rc.invulnerables, pallet_staking_async::Invulnerables::<T>::get());
-		assert_equal_items(
-			rc.bonded.into_iter().map(|(a, b)| (t(a), t(b))),
-			pallet_staking_async::Bonded::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.ledger.into_iter().map(|(k, v)| (t(k), v.translate_accounts(&t).into())),
-			pallet_staking_async::Ledger::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.payee.into_iter().map(|(k, v)| (t(k), translate_reward_destination(v, &t))),
-			pallet_staking_async::Payee::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.validators.into_iter().map(|(k, v)| (k, translate_validator_prefs(v))),
-			pallet_staking_async::Validators::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.nominators.into_iter().map(|(k, v)| (t(k), translate_nominations(v, &t))),
-			pallet_staking_async::Nominators::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.virtual_stakers.into_iter().map(t),
-			pallet_staking_async::VirtualStakers::<T>::iter_keys(),
-		);
-		assert_equal_items(
-			rc.eras_stakers_overview.into_iter().map(|(k1, k2, v)| (k1, t(k2), v.into())),
-			pallet_staking_async::ErasStakersOverview::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.eras_stakers_paged
-				.into_iter()
-				.map(|((k0, k1, k2), v)| ((k0, t(k1), k2), v.translate_accounts(&t).into())),
-			pallet_staking_async::ErasStakersPaged::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.claimed_rewards
-				.into_iter()
-				.map(|(k0, k1, v)| (k0, t(k1), WeakBoundedVec::force_from(v, None))),
-			pallet_staking_async::ClaimedRewards::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.eras_validator_prefs.into_iter().map(|(k1, k2, v)| (k1, t(k2), v.into())),
-			pallet_staking_async::ErasValidatorPrefs::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.eras_validator_reward,
-			pallet_staking_async::ErasValidatorReward::<T>::iter(),
-		);
-		assert_equal_items(
-			rc.eras_reward_points
-				.into_iter()
-				.map(|(k, v)| (k, v.translate_accounts(&t).into())),
-			pallet_staking_async::ErasRewardPoints::<T>::iter(),
-		);
-		assert_equal_items(rc.eras_total_stake, pallet_staking_async::ErasTotalStake::<T>::iter());
-		check_unapplied_slashes::<T>(rc.unapplied_slashes, &t);
-		assert_equal_items(rc.bonded_eras, pallet_staking_async::BondedEras::<T>::get());
-		assert_equal_items(
-			rc.validator_slash_in_era.into_iter().map(|(k0, k1, v)| (k0, t(k1), v)),
-			pallet_staking_async::ValidatorSlashInEra::<T>::iter(),
-		);
-	}
+    type RcPrePayload = RcData;
+    type AhPrePayload = ();
+
+    fn pre_check(_rc: Self::RcPrePayload) -> Self::AhPrePayload {}
+
+    fn post_check(rc: Self::RcPrePayload, _ah_pre_payload: Self::AhPrePayload) {
+        let t = crate::Pallet::<T>::translate_account_rc_to_ah;
+
+        // Storage Values
+        assert_eq!(
+            rc.validator_count,
+            pallet_staking_async::ValidatorCount::<T>::get()
+        );
+        // Min validator count is not migrated and instead configured via `MinimumValidatorSetSize`
+        assert_eq!(
+            rc.min_nominator_bond,
+            pallet_staking_async::MinNominatorBond::<T>::get()
+        );
+        assert_eq!(
+            rc.min_validator_bond,
+            pallet_staking_async::MinValidatorBond::<T>::get()
+        );
+        assert_eq!(
+            rc.min_active_stake,
+            pallet_staking_async::MinimumActiveStake::<T>::get()
+        );
+        assert_eq!(
+            rc.min_commission,
+            pallet_staking_async::MinCommission::<T>::get()
+        );
+        assert_eq!(
+            rc.max_validators_count,
+            pallet_staking_async::MaxValidatorsCount::<T>::get()
+        );
+        assert_eq!(
+            rc.max_nominators_count,
+            pallet_staking_async::MaxNominatorsCount::<T>::get()
+        );
+        assert_eq!(
+            pallet_staking_async::CurrentEra::<T>::get().expect("Must be set"),
+            pallet_staking_async::ActiveEra::<T>::get()
+                .expect("Must be set")
+                .index
+        );
+        assert_eq!(
+            rc.active_era.map(translate_active_era),
+            pallet_staking_async::ActiveEra::<T>::get()
+        );
+        assert_eq!(
+            translate_forcing(rc.force_era),
+            pallet_staking_async::ForceEra::<T>::get()
+        );
+        assert_eq!(
+            rc.max_staked_rewards,
+            pallet_staking_async::MaxStakedRewards::<T>::get()
+        );
+        assert_eq!(
+            rc.slash_reward_fraction,
+            pallet_staking_async::SlashRewardFraction::<T>::get()
+        );
+        assert_eq!(
+            rc.canceled_slash_payout,
+            pallet_staking_async::CanceledSlashPayout::<T>::get()
+        );
+        // Current planned session is not migrated
+        assert_eq!(
+            rc.chill_threshold,
+            pallet_staking_async::ChillThreshold::<T>::get()
+        );
+
+        // Storage Maps
+        assert_equal_items(
+            rc.invulnerables,
+            pallet_staking_async::Invulnerables::<T>::get(),
+        );
+        assert_equal_items(
+            rc.bonded.into_iter().map(|(a, b)| (t(a), t(b))),
+            pallet_staking_async::Bonded::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.ledger
+                .into_iter()
+                .map(|(k, v)| (t(k), v.translate_accounts(&t).into())),
+            pallet_staking_async::Ledger::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.payee
+                .into_iter()
+                .map(|(k, v)| (t(k), translate_reward_destination(v, &t))),
+            pallet_staking_async::Payee::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.validators
+                .into_iter()
+                .map(|(k, v)| (k, translate_validator_prefs(v))),
+            pallet_staking_async::Validators::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.nominators
+                .into_iter()
+                .map(|(k, v)| (t(k), translate_nominations(v, &t))),
+            pallet_staking_async::Nominators::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.virtual_stakers.into_iter().map(t),
+            pallet_staking_async::VirtualStakers::<T>::iter_keys(),
+        );
+        assert_equal_items(
+            rc.eras_stakers_overview
+                .into_iter()
+                .map(|(k1, k2, v)| (k1, t(k2), v.into())),
+            pallet_staking_async::ErasStakersOverview::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.eras_stakers_paged
+                .into_iter()
+                .map(|((k0, k1, k2), v)| ((k0, t(k1), k2), v.translate_accounts(&t).into())),
+            pallet_staking_async::ErasStakersPaged::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.claimed_rewards
+                .into_iter()
+                .map(|(k0, k1, v)| (k0, t(k1), WeakBoundedVec::force_from(v, None))),
+            pallet_staking_async::ClaimedRewards::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.eras_validator_prefs
+                .into_iter()
+                .map(|(k1, k2, v)| (k1, t(k2), v.into())),
+            pallet_staking_async::ErasValidatorPrefs::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.eras_validator_reward,
+            pallet_staking_async::ErasValidatorReward::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.eras_reward_points
+                .into_iter()
+                .map(|(k, v)| (k, v.translate_accounts(&t).into())),
+            pallet_staking_async::ErasRewardPoints::<T>::iter(),
+        );
+        assert_equal_items(
+            rc.eras_total_stake,
+            pallet_staking_async::ErasTotalStake::<T>::iter(),
+        );
+        check_unapplied_slashes::<T>(rc.unapplied_slashes, &t);
+        assert_equal_items(rc.bonded_eras, pallet_staking_async::BondedEras::<T>::get());
+        assert_equal_items(
+            rc.validator_slash_in_era
+                .into_iter()
+                .map(|(k0, k1, v)| (k0, t(k1), v)),
+            pallet_staking_async::ValidatorSlashInEra::<T>::iter(),
+        );
+    }
 }
 
 #[allow(deprecated)]
 fn translate_reward_destination(
-	destination: pallet_staking::RewardDestination<AccountId32>,
-	t: &impl Fn(AccountId32) -> AccountId32,
+    destination: pallet_staking::RewardDestination<AccountId32>,
+    t: &impl Fn(AccountId32) -> AccountId32,
 ) -> pallet_staking_async::RewardDestination<AccountId32> {
-	use pallet_staking_async::RewardDestination::*;
-
-	match destination {
-		pallet_staking::RewardDestination::Staked => Staked,
-		pallet_staking::RewardDestination::Stash => Stash,
-		pallet_staking::RewardDestination::Controller => Controller,
-		pallet_staking::RewardDestination::Account(account) => Account(t(account)),
-		pallet_staking::RewardDestination::None => None,
-	}
+    use pallet_staking_async::RewardDestination::*;
+
+    match destination {
+        pallet_staking::RewardDestination::Staked => Staked,
+        pallet_staking::RewardDestination::Stash => Stash,
+        pallet_staking::RewardDestination::Controller => Controller,
+        pallet_staking::RewardDestination::Account(account) => Account(t(account)),
+        pallet_staking::RewardDestination::None => None,
+    }
 }
 
 fn translate_active_era(era: pallet_staking::ActiveEraInfo) -> pallet_staking_async::ActiveEraInfo {
-	pallet_staking_async::ActiveEraInfo { index: era.index, start: era.start }
+    pallet_staking_async::ActiveEraInfo {
+        index: era.index,
+        start: era.start,
+    }
 }
 
 fn translate_forcing(forcing: pallet_staking::Forcing) -> pallet_staking_async::Forcing {
-	use pallet_staking_async::Forcing;
-	match forcing {
-		pallet_staking::Forcing::NotForcing => Forcing::NotForcing,
-		pallet_staking::Forcing::ForceNew => Forcing::ForceNew,
-		pallet_staking::Forcing::ForceNone => Forcing::ForceNone,
-		pallet_staking::Forcing::ForceAlways => Forcing::ForceAlways,
-	}
+    use pallet_staking_async::Forcing;
+    match forcing {
+        pallet_staking::Forcing::NotForcing => Forcing::NotForcing,
+        pallet_staking::Forcing::ForceNew => Forcing::ForceNew,
+        pallet_staking::Forcing::ForceNone => Forcing::ForceNone,
+        pallet_staking::Forcing::ForceAlways => Forcing::ForceAlways,
+    }
 }
 
 fn translate_validator_prefs(
-	prefs: pallet_staking::ValidatorPrefs,
+    prefs: pallet_staking::ValidatorPrefs,
 ) -> pallet_staking_async::ValidatorPrefs {
-	pallet_staking_async::ValidatorPrefs { commission: prefs.commission, blocked: prefs.blocked }
+    pallet_staking_async::ValidatorPrefs {
+        commission: prefs.commission,
+        blocked: prefs.blocked,
+    }
 }
 
 fn translate_nominations<T: crate::Config>(
-	nominations: PortableNominations,
-	t: &impl Fn(AccountId32) -> AccountId32,
+    nominations: PortableNominations,
+    t: &impl Fn(AccountId32) -> AccountId32,
 ) -> pallet_staking_async::Nominations<T> {
-	pallet_staking_async::Nominations {
-		targets: nominations
-			.targets
-			.into_inner()
-			.into_iter()
-			.map(t)
-			.collect::<Vec<_>>()
-			.try_into()
-			.expect("Must not truncate"),
-		submitted_in: nominations.submitted_in,
-		suppressed: nominations.suppressed,
-	}
+    pallet_staking_async::Nominations {
+        targets: nominations
+            .targets
+            .into_inner()
+            .into_iter()
+            .map(t)
+            .collect::<Vec<_>>()
+            .try_into()
+            .expect("Must not truncate"),
+        submitted_in: nominations.submitted_in,
+        suppressed: nominations.suppressed,
+    }
 }
 
 fn check_unapplied_slashes<T: crate::Config>(
-	rc: Vec<(u32, Vec<PortableUnappliedSlash>)>,
-	t: &impl Fn(AccountId32) -> AccountId32,
+    rc: Vec<(u32, Vec<PortableUnappliedSlash>)>,
+    t: &impl Fn(AccountId32) -> AccountId32,
 ) {
-	let mut expected_slashes = Vec::new();
+    let mut expected_slashes = Vec::new();
 
-	for (era, slashes) in rc {
-		for slash in slashes {
-			// We insert all slashes with this special key
-			let key = (t(slash.clone().validator), Perbill::from_percent(99), 9999);
-			expected_slashes.push((era, key, slash.translate_accounts(t).into()));
-		}
-	}
+    for (era, slashes) in rc {
+        for slash in slashes {
+            // We insert all slashes with this special key
+            let key = (t(slash.clone().validator), Perbill::from_percent(99), 9999);
+            expected_slashes.push((era, key, slash.translate_accounts(t).into()));
+        }
+    }
 
-	assert_equal_items(expected_slashes, pallet_staking_async::UnappliedSlashes::<T>::iter());
+    assert_equal_items(
+        expected_slashes,
+        pallet_staking_async::UnappliedSlashes::<T>::iter(),
+    );
 }
 
 /// Assert that two iterators have the same elements, regardless of their order.
 fn assert_equal_items<
-	V: Encode + PartialEq + Debug,
-	I: IntoIterator<Item = V>,
-	J: IntoIterator<Item = V>,
+    V: Encode + PartialEq + Debug,
+    I: IntoIterator<Item = V>,
+    J: IntoIterator<Item = V>,
 >(
-	rc: I,
-	ah: J,
+    rc: I,
+    ah: J,
 ) {
-	let mut rc: Vec<V> = rc.into_iter().collect::<Vec<_>>();
-	rc.sort_by_encoded();
-	let mut ah: Vec<V> = ah.into_iter().collect::<Vec<_>>();
-	ah.sort_by_encoded();
-
-	for (i, (r, a)) in rc.iter().zip(ah.iter()).enumerate() {
-		assert_eq!(r, a, "Entry #{i} mismatch: {r:?} != {a:?}");
-	}
+    let mut rc: Vec<V> = rc.into_iter().collect::<Vec<_>>();
+    rc.sort_by_encoded();
+    let mut ah: Vec<V> = ah.into_iter().collect::<Vec<_>>();
+    ah.sort_by_encoded();
+
+    for (i, (r, a)) in rc.iter().zip(ah.iter()).enumerate() {
+        // assert_eq!(r, a, "Entry #{i} mismatch: {r:?} != {a:?}");
+    }
 }
diff --git a/pallets/rc-migrator/src/scheduler.rs b/pallets/rc-migrator/src/scheduler.rs
index 581965b7f..4e8b092fa 100644
--- a/pallets/rc-migrator/src/scheduler.rs
+++ b/pallets/rc-migrator/src/scheduler.rs
@@ -378,10 +378,10 @@ impl<T: Config> crate::types::RcMigrationCheck for SchedulerMigrator<T> {
 
 	fn post_check(_rc_pre_payload: Self::RcPrePayload) {
 		// Assert storage 'Scheduler::IncompleteSince::rc_post::empty'
-		assert!(
-			pallet_scheduler::IncompleteSince::<T>::get().is_none(),
-			"IncompleteSince should be None on RC after migration"
-		);
+		// assert!(
+		// pallet_scheduler::IncompleteSince::<T>::get().is_none(),
+		// "IncompleteSince should be None on RC after migration"
+		// );
 
 		// Assert storage 'Scheduler::Agenda::rc_post::empty'
 		assert!(
```

To be noted for nomination-pools we see the following:
```bash
2025-09-26T10:11:05.228866Z  INFO runtime::frame-support: 🩺 Running "NominationPools" try-state checks    
2025-09-26T10:11:05.229601Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 196: 0 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.229637Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 54: 3 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.229774Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 37: 1 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230045Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 188: 3 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230219Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 198: 16 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230264Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 202: 0 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230341Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 192: 0 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230413Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 132: 18 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230546Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 119: 0 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230699Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 159: 0 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230743Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 69: 2 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230885Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 46: 2 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.230930Z  WARN runtime::nomination-pools: [11008326] 🏊‍♂️ reward pool of 174: 0 (ed = 3333333), should only happen because ED has changed recently. Pool operators should be notified to top up the reward account    
2025-09-26T10:11:05.328431Z ERROR runtime::frame-support: ❌ "NominationPools" try_state checks failed: Other("Some pools do not have correct ED frozen") 
```

@kianenigma  are we fine with that? As discussed already, we haven't backported SDK #9415 to 2507, are we fine with that for KAHM?
